src/broker/log.c

```diff
@@ -76,11 +76,12 @@ static struct logbuf_entry *logbuf_entry_create (const char *buf, int len)
     struct logbuf_entry *e = calloc (1, sizeof (*e));
     if (!e)
         return NULL;
-    if (!(e->buf = calloc (1, len))) {
+    if (!(e->buf = malloc (len + 1))) {
         free (e);
         return NULL;
     }
     memcpy (e->buf, buf, len);
+    e->buf[len] = '\0'; // stealth NULL string termination
```

Comment:
i think this `NULL` should be `NUL`

Reply:
OK, please explain "nul" versus "null"?   In C it's NULL and my dictionary likes the word spelled as "null" not "nul".   I've always used "NULL" or "null" here, but I'm happy to learn.

Reply:
One less "L"

Reply:
Oh, so we should drop al duplicate characters?

Reply:
In all seriousness I always assumed `NUL` was an ascii thing. Here's a SO post with reasonable answers.  I think you can probably use what you want here since the meaning is implied

https://stackoverflow.com/questions/1296843/what-is-the-difference-between-null-0-and-0

Reply:
we can save even more space if we sort the characters, *then* drop duplicates

Reply:
OK I fixed the comment, choosing words that would be less controversial.
Also dropped `e->len` since it wasn't being used anymore.

Reply:
> In all seriousness I always assumed NUL was an ascii thing

Oh OK.  For some reason that didn't occur to me :-)

Reply:
I did not know there was debate / ambiguity about this.  I always thought it was NULL for pointers and NUL for ascii `\0`.  A simple "NULL vs NUL" in google pointed me to this:

https://stackoverflow.com/questions/30121241/what-is-the-difference-between-nul-and-null

which may not be the most definitive answer.  But I've always operated on this.